routesrv/routesrv.go

```diff
@@ -91,6 +115,22 @@ func New(opts Options) (*RouteServer, error) {
 		oauthConfig.CallbackPath = opts.OAuth2CallbackPath
 	}
 
+	var rh *RedisHandler
+	// in case we have kubernetes dataclient and we can detect redis instances, we patch redisOptions
+	if opts.KubernetesRedisServiceNamespace != "" && opts.KubernetesRedisServiceName != "" {
+		log.Infof("Use endpoints %s/%s to fetch updated redis shards", opts.KubernetesRedisServiceNamespace, opts.KubernetesRedisServiceName)
+		rh = &RedisHandler{}
+		rh.AddrUpdater = getRedisAddresses(opts.KubernetesRedisServiceNamespace, opts.KubernetesRedisServiceName, dataclient)
+		handler.Handle("/swarm", rh)
```

Comment:
Let's add another line with:
```
dataclient.LoadAll()
```
to make sure that we have an initial clusterstate and drop the comment in line 43

Reply:
Thanks was struggling with the testcase